### PR TITLE
Implement Circonus IRONdb origin type

### PR DESCRIPTION
**Pull request by dhaifley**
- Adds an origin type for IRONdb databases supporting all of the IRONdb data retrieval API's.
- This allows Trickster to be used as a data source when using the Grafana IRONdb datasource plug in.
- Updates some documentation to reflect the inclusion of the IRONdb origin type and describe its usage.

### CODE REVIEW

**Comment by jranson** [comment]
@dhaifley  Nice!  Thanks David! I will review this today. In the mean time, if you don't mind completing the CLA process, that should square everything away on the paperwork side. Appreciate you keeping up with all the breaking merges on this as well. We're very appreciative. -JR

**Comment by dhaifley** [comment]
@jranson Will do ASAP.  Since I'm signing on behalf of Circonus, I'm just waiting to get the ok from my boss.

**Comment by jranson** [comment]
@dhaifley Everything looks good except for one little nit:

```
gofmt: 99%
	internal/proxy/origins/irondb/model_df4_test.go
		Line 1: warning: file is not gofmted with -s (gofmt)
```

You should be able to fix by running `gofmt -w -s /your/local/path/to/model_df4_test.go` and repushing.

File: .gitignore [hunk 1]
```diff
@@ -42,5 +42,8 @@ cacheKey.expiration
 #Goland IDE
 .idea/
 
+# VSCode editor
```
Comment (jranson) [inline .gitignore]: Would you remind moving this section? You'll find we actually already added this one in on line 17 as part of the most recent merge. GMTA :)

Comment (dhaifley) [reply .gitignore]: Thanks James.  I've made the requested changes.

**Comment by dhaifley** [comment]
It took a while for our legal review, but the CLA is now signed. Thanks!

**Pull request merged**